lib/Service/PhishingDetectionService.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * @copyright 2024 Hamza Mahjoubi <hamza.mahjoubi221@proton.me>
+ *
+ * @author 2024 Hamza Mahjoubi <hamza.mahjoubi221@proton.me>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Mail\Service;
+
+use DateTime;
+use Horde_Mime_Headers;
+use OCA\Mail\AddressList;
+use OCA\Mail\Contracts\ITrustedSenderService;
+
+class PhishingDetectionService {
+
+
+	private ContactsIntegration $contactIntegration;
+
+	private ITrustedSenderService $trustedSenderService;
+
+
+	private bool $warn = false;
+
+
+	public function __construct(ContactsIntegration $contactIntegration, ITrustedSenderService $trustedSenderService) {
+		$this->contactIntegration = $contactIntegration;
+		$this->trustedSenderService = $trustedSenderService;
+	}
+
+	private function checkDatePass(string $date): array {
```

Comment:
I would suggest to name all checking methods in a similar way to make this more comprehensible. Some have "check" in the beginning, some at the end. And what does the "Pass"-suffix on some methods mean?